docs/extensions.adoc

```diff
@@ -900,6 +900,9 @@ and also attached one annotation directly.
 Repeatable annotations are not supported for parameter annotations,
 as they are primarily intended to provide a value and only a singular value is allowed.
 
+NOTE: Since Spock 2.4 the new subtype `IStatelessAnnotationDrivenExtension` is available, which should be used if the extension does not need to store Specification specific state.
```

Comment:
I would expect that the extension also need to be `thread-safe` in addition to not using Specification state.

Reply:
How can it be not thread-safe if it is stateless? In any case, the discovery and spec configuration is single-threaded.

Reply:
No, I meant to add a sentence that an user implemented extension need also be thread-safe.
One could miss interpret the "does not need to store Specification specific state.", that you can have other state somehow...